readthedocs/builds/models.py

```diff
@@ -1212,3 +1222,29 @@ def get_edit_url(self):
             'projects_automation_rule_regex_edit',
             args=[self.project.slug, self.pk],
         )
+
+
+class AutomationRuleMatch(TimeStampedModel):
+    rule = models.ForeignKey(
+        VersionAutomationRule,
+        verbose_name=_('Automation rule match'),
+        related_name='matches',
+        on_delete=models.CASCADE,
+    )
+
+    # Metadata from when the match happened.
+    version_name = models.CharField(max_length=255)
+    match_arg = models.CharField(max_length=255)
+    action = models.CharField(
```

Comment:
Are we duplicating this here in case the user modified the `VersionAutomationRule` we have linked via FK?

Reply:
yeah, we need to save the data from when the match happened.